packages/metadata/src/__tests__/unit/decorator-factory.unit.ts

```diff
@@ -80,6 +80,12 @@ describe('ClassDecoratorFactory', () => {
     return ClassDecoratorFactory.createDecorator('test', spec);
   }
 
+  function testDecorator(spec: object): ClassDecorator {
+    return ClassDecoratorFactory.createDecorator('test', spec, {
```

Comment:
same question here, `test` and `@test` seems duplicate, and what would happen if  they are different?

Reply:
The first argument is the key for the metadata, it can be `test` or `test.key`, or other unique strings. The second one is for the decorator function name.